connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/http/InstanceForwardingHttpClient.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.runtime.core.http;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.connector.api.json.ConnectorsObjectMapperSupplier;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+
+public class InstanceForwardingHttpClient {
+  private static final ObjectMapper OBJECT_MAPPER = ConnectorsObjectMapperSupplier.getCopy();
+  private final HttpClient httpClient;
+  private final InstancesUrlBuilder urlBuilder;
```

Comment:
Maybe I am missing something but we couple the forwarding aspect with the fetching the data aspect here.

What about having one service that knows the instances and creates the URLs and does the calls just a regular RestClient in a loop or in parallel using a Flux.

Reply:
Example as an inspiration:
```
Flux.fromIterable(ids)
  .flatMap(id -> webClient.get()
    .uri("/comments/{id}", id)
    .accept(MediaType.APPLICATION_JSON)
    .retrieve()
    .bodyToMono(Comment.class))
  .subscribeOn(Schedulers.parallel());
```
Source: https://stackoverflow.com/questions/51118848/springboot-webflux-making-parallel-webclient-requests

Reply:
@johnBgood What if the `AggregatingInboundInstanceService` is the one building the URLs and just issuing the HTTP requests?

Reply:
I was just wondering why we need this extra http client and not have a simple Interface that implements the actual http calls? Can be hidden behin an interface like `InboundInstanceApi`